util/builtinobjects/builtinobjects.go

```diff
@@ -139,7 +139,7 @@ func (b *builtinObjects) inject(ctx *session.Context, archive []byte, isMigratio
 
 	// TODO: GO-1387 Need to use profile.pb to handle dashboard injection during migration
 	oldId := migrationDashboardName
-	if !isMigration {
+	if useCase != migrationUseCase {
```

Comment:
Where can useCase be set to -1?

Reply:
Nowhere, that's why I used it as corner-case for migration